client/network/src/error.rs

```diff
@@ -48,7 +48,9 @@ pub enum Error {
 		second_id: PeerId,
 	},
 	/// Prometheus metrics error.
-	Prometheus(prometheus_endpoint::PrometheusError)
+	Prometheus(prometheus_endpoint::PrometheusError),
+	/// Invalid configuration.
```

Comment:
I know that this pattern of using Strings for error messages is common in Substrate, but to me it is a bad programming practice nonetheless.
Please change to an enum, or at least mention what the string is.

```suggestion
	/// Invalid configuration. Contains a human-readable error message.
```

Reply:
Also, `Error::DuplicateBootnode` is normally part of `InvalidConfiguration` as well.
I know I'm being a pain in the ass, but I'm trying to remove quick hacks from `sc-network` as much as possible.

Reply:
Not at all!! This is fine, I will improve it! Thanks for the feedback

Reply:
@tomaka voila. I made the error more "generic". It's a bit less descriptive but I think it's still good. I think it would have been weird to have 2 different errors to make the distinction in the human readable text.